pkgs/development/libraries/webkitgtk/default.nix

```diff
@@ -63,20 +64,45 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" ];
 
-  separateDebugInfo = stdenv.isLinux;
+  separateDebugInfo = stdenv.hostPlatform.isLinux;
 
   src = fetchurl {
     url = "https://webkitgtk.org/releases/${pname}-${version}.tar.xz";
     sha256 = "07vzbbnvz69rn9pciji4axfpclp98bpj4a0br2z0gbn5wc4an3bx";
   };
 
-  patches = optionals stdenv.isLinux [
+  patches = optionals stdenv.hostPlatform.isLinux [
     (substituteAll {
       src = ./fix-bubblewrap-paths.patch;
       inherit (builtins) storeDir;
     })
     ./libglvnd-headers.patch
-  ];
+    ./darwin/patch-WTF-wtf-Randomdevice.diff
```

Comment:
Could you add a comment to each patch explaining why it is needed, what it does and where does it come from? Ideally, we would use `fetchpatch` to get them.

Reply:
Yes, I will once I get it building. These patches are all taken from the MacPort of nyxt browser (https://source.atlas.engineer/view/repository/macports-port), however, the maintainer of the MacPort will not be continuing packaging it, so I put this in tree until we either get them upstream which the WebKitGTK maintainers are amenable to (these are for MacOS compatibility not nyxt) or find another home for them where we can keep them up to date.